src/components/Table/components/ExtendedColumnItem/ExtendedColumnItem.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import useTheme from 'hooks/useTheme';
+import { ExtendedColumn, Sort } from '../../types';
+import { containerStyles, contentStyles } from './ExtendedColumnItem.style';
+import Icon from 'components/Icon';
+import Tooltip from 'components/Tooltip';
+
+type Props = {
+  item: ExtendedColumn | string;
+  isNumerical?: boolean;
+  sorting?: Sort;
+};
+
+const ExtendedColumnItem: React.FC<Props> = ({ item, sorting, isNumerical }) => {
+  const theme = useTheme();
+
+  const sortingItem = () =>
+    //TODO: Remove type check when backwards-compatibility is removed
+    typeof item !== 'string' &&
```

Comment:
same thing here for `typeof item !== 'string'` define it

Reply:
another thing is to create a utility under table with a function `const isPropTypeString = (prop) = typeof prop === 'string'` and you can use this to define it to a const `const isItemTypeString = isPropTypeString(item);`